sdk/python/kfp/dsl/_container_op.py

```diff
@@ -887,18 +893,37 @@ def add_pod_label(self, name: str, value: str):
     self.pod_labels[name] = value
     return self
 
-  def set_retry(self, num_retries: int, policy: str = None):
+  def set_retry(self,
+                num_retries: int,
+                policy: Optional[str] = None,
+                backoff_duration: Optional[str] = None,
+                backoff_factor: Optional[float] = None,
+                backoff_max_duration: Optional[str] = None):
     """Sets the number of times the task is retried until it's declared failed.
 
     Args:
       num_retries: Number of times to retry on failures.
       policy: Retry policy name.
+      backoff_duration: The time interval between retries. Defaults to an
+        immediate retry. In case you specify a simple number, the unit
+        defaults to seconds. You can also specify a different unit, for
+        instance, 2m (2 minutes), 1h (1 hour).
+      backoff_factor: The exponential backoff factor applied to
+        backoff_duration. For example, if backoff_duration="60"
+        (60 seconds) and backoff_factor=2, the first retry will happen
+        after 60 seconds, then after 120, 240, and so on.
+      backoff_max_duration: The maximum interval that can be reached with
+        the backoff strategy.
     """
     if policy is not None and policy not in ALLOWED_RETRY_POLICIES:
-      raise ValueError('policy must be one of: %r' % (ALLOWED_RETRY_POLICIES,))
+        raise ValueError('policy must be one of: %r'
```

Comment:
Just spotted that there's an error indentation.

This PR was created before I changed the indentation to 2 spaces in this file (https://github.com/kubeflow/pipelines/pull/5226), so when it was rebased on top of the head, it reserved this diff. I'll fix this in my next PR touching this file.

Reply:
Ouch, sorry @chensun indeed during the rebase I had to fix the indentation of my changes but completely missed this one.